Automattic/wp-calypso
# Domains: Allow Multi-Year Domain Registration

### Summary
This PR allows users to register a domain for multiple years by rendering a dropdown on the Checkout page.
• Adding support for plans and terms up to 10 years (decennial)
• Adding support for differentiating products based on volume

![image](https://github.com/Automattic/wp-calypso/assets/20927667/b47425ad-855e-426f-acf4-1e0e491df4b4)

PT: pau2Xa-4UB-p2

### Testing Instructions

#### Prerequisites
1. Add `define( 'USE_STORE_SANDBOX', true );` to `wp-content/mu-plugins/0-sandbox.php` to sandbox the store so you can checkout with a test credit card
2. Sandbox `public-api.wordpress.com`
3. Apply D116538-code by running `arc patch D116538` on your sandbox
4. Checkout this branch
5. Run `yarn` and `yarn start`

#### Testing
1. Go to http://calypso.localhost:3000/start
2. Select a custom `.com` domain
3. Proceed to the Checkout page
5. For the Domain Registration product, confirm you see multiple options for the number of years the domain can be registered.
6. Open the `Network` tab and filter the transactions by `me/shopping-cart`
7. Click on the dropdown and choose a different option
8. Expand the Payload of the corresponding network request and verify the correct `volume` is being passed to the BE: `volume: 1` for **One Year**,`volume: 2` for **Two Years** and so on. Expand the Response and verify the correct `volume` is returned from the BE.
9. Verify the selection is reflected in the dropdown
10. Verify the selection is reflected in the dropdown on the Checkout page
11. Open the `Network` tab and filter the transactions by `transactions`
12. Use a test credit card to complete the purchase
13. Expand the Payload of the corresponding network request and verify the correct `volume` is being passed to the BE:
`volume: 1` for **One Year**,`volume: 2` for **Two Years** and so on.

#### Testing Video
https://github.com/Automattic/wp-calypso/assets/20927667/bece5d8d-de8a-4d9d-b3d7-4958b02efa9f
